### PR TITLE
docs(installation): update installation pages

### PR DESCRIPTION
**Pull request by shmpwk**
Signed-off-by: Shumpei Wakabayashi <shumpei.wakabayashi@tier4.jp>

## Pre-review checklist for the PR author

In the [Installation page](https://autowarefoundation.github.io/autoware-documentation/main/installation/) of Autoware Documentation, "Tools for developers" might mislead readers since the contents are not actually Tools.
So I suggest changing the name of the pages.

Related discussion:
https://github.com/autowarefoundation/autoware/discussions/280

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/installation/index.md
Comment (kenji-miyake) [inline docs/installation/index.md:25]: If you change here, I think it's better to write like this.

```md
## Installation steps

### Autoware

### Tools for users
```

Comment (kenji-miyake) [reply docs/installation/index.md:25]: Also, if there is only one word "tools", I think we can change this section as well.

```suggestion
## Related tools
```

Or this?
```suggestion
## Peripheral tools
```

@LalithVipulananthan Any ideas?

Comment (shmpwk) [reply docs/installation/index.md]: Thank you for your review and almost agree with you.

I think shallow hierarchy is more simpler?
like
```
- Installation
  - Installation steps of Autoware
  - Installation steps of related tools
  - Additional settings for developer
```
than 
```
- Installation
  - Installation steps 
    -  Autoware
    - Related tools
  - Additional settings for developer
```

Comment (LalithVipulananthan) [reply docs/installation/index.md]: A shallow hierarchy seems fine, but `Installation steps of Autoware` and `Installation steps of related tools` should be shortened to `Autoware installation` and `Related tools` IMO.

On a related note, I think that the Autoware installation page is so short that we could perhaps merge it with the index page? 🤔

Comment (shmpwk) [reply docs/installation/index.md]: > A shallow hierarchy seems fine, but Installation steps of Autoware and Installation steps of related tools should be shortened to Autoware installation and Related tools IMO.

I agree!


> we could perhaps merge it with the index page? 

You mean [installation/index.md#autoware-installation](https://github.com/autowarefoundation/autoware-documentation/blob/f3b835552193a1f5c816d98741b2f00bd702ac24/docs/installation/index.md#autoware-installation) section is better to be merged with [installation/autoware/index.md](https://github.com/autowarefoundation/autoware-documentation/blob/f3b835552193a1f5c816d98741b2f00bd702ac24/docs/installation/autoware/index.md).

Yes, I agree it is simpler.

File: docs/installation/index.md
Comment (kenji-miyake) [inline docs/installation/index.md:40]: If you write this here, it's better to remove `installation/autoware/index.md`.
It seems duplicated now.

https://autowarefoundation.github.io/autoware-documentation/pr-104/installation/autoware/

Comment (shmpwk) [reply docs/installation/index.md:40]: Ok, I removed the duplication.

Note that the botton of "Autoware" don"t work. 
Do you think it is ok?

Comment (kenji-miyake) [reply docs/installation/index.md:40]: Yes, I think it's okay. Thank you!

**Review by kenji-miyake — DISMISSED** [review]
Please wait for @LalithVipulananthan -san's approval as well. :bow:

File: docs/installation/index.md
Comment (LalithVipulananthan) [inline docs/installation/index.md:31]: @kenji-miyake On reflection, I feel that the Docker README file with example run commands should be added to the Autoware documentation. What do you think about this idea?

File: docs/installation/index.md
Comment (LalithVipulananthan) [inline docs/installation/index.md:40]: The autowarefoundation/autoware/README.md has been changed quite significantly and no longer contains any information that is relevant to installing from source, so I think this line should be removed.

File: docs/installation/index.md
Comment (LalithVipulananthan) [inline docs/installation/index.md:39]: Meaningful link text:

```suggestion
Refer to the [source installation guide](https://github.com/autowarefoundation/autoware-documentation/blob/main/docs/installation/autoware/source-installation.md) for detailed steps.
```

File: docs/installation/index.md
Comment (LalithVipulananthan) [inline docs/installation/index.md:37]: @kenji-miyake Although in theory it is good to have this line as cover in case someone's environment breaks after installing Autoware, in practice I feel that it is too vague to be helpful. It also suggests that the installation process is not stable which is not true!

Could we remove this line perhaps?

File: docs/installation/index.md
Comment (LalithVipulananthan) [inline docs/installation/index.md:45]: It may not be obvious to someone what `E2E` means and it is only referred to once on this page, so it is better to write out that in full.

```suggestion
For example, to run an end-to-end simulation you need to install an appropriate simulator.
```

File: docs/installation/index.md
Comment (LalithVipulananthan) [inline docs/installation/index.md:42]: This was Miyake-san's suggestion for the title of this section and I agree that it reads better.

```suggestion
## Related tools
```

**Review by LalithVipulananthan — COMMENTED** [review]
@shmpwk As requested by Miyake-san, I have reviewed the PR. Please let me know if you have any questions about my comments.

File: docs/installation/index.md
Comment (shmpwk) [reply docs/installation/index.md]: Thank you.
Fixed by https://github.com/autowarefoundation/autoware-documentation/pull/104/commits/23ef5d26c46d4e75a9d157867a402280992d266c

File: docs/installation/index.md
Comment (kenji-miyake) [reply docs/installation/index.md:31]: @LalithVipulananthan Hmm? I guess it's here. Could you explain it a bit more in detail?
https://github.com/autowarefoundation/autoware-documentation/blob/9834c0e3d525648f3bcef1d22c8649b7e3386eb9/docs/installation/autoware/docker-installation.md?plain=1#L60

File: docs/installation/index.md
Comment (kenji-miyake) [reply docs/installation/index.md:37]: @LalithVipulananthan How about writing like "Refer to the troubleshooting page when you occur a problem."?

Comment (LalithVipulananthan) [reply docs/installation/index.md:37]: This seems like a good compromise. Since installation problems could occur with Docker as well, could we add this suggestion further up the page around line 23? For example:

> There are two ways to set up Autoware. Choose one according to your preference. 
> If any issues occur during installation, refer to the Troubleshooting guide.

File: docs/installation/index.md
Comment (LalithVipulananthan) [reply docs/installation/index.md:31]: I understand the Docker installation page has a link to the advanced usage commands that are in the README of autowarefoundation/autoware/docker, but I just felt that for consistency these commands should be part of the core documentation. It's not that big a deal so if you think that information is fine where it is then that's okay! 😄 

That said, if the Docker installation page contains a reference to the README.md file, then we don't need to have a link to it on this page as well right?

File: docs/installation/index.md
Comment (shmpwk) [reply docs/installation/index.md]: I agree!

File: docs/installation/index.md
Comment (kenji-miyake) [reply docs/installation/index.md:31]: > but I just felt that for consistency these commands should be part of the core documentation.

I think details should be in `autoware/docker/README.md` for easy maintenance and that the summary should be in autoware-documentation.

> That said, if the Docker installation page contains a reference to the README.md file, then we don't need to have a link to it on this page as well right?

Right, we can remove the duplicated link from this page.

File: docs/installation/index.md
Comment (kenji-miyake) [reply docs/installation/index.md:37]: Yes, it's nice. :+1: 
@shmpwk Could you please update the documents as per Lalith-san's suggestion? :pray:

File: docs/installation/index.md
Comment (kenji-miyake) [reply docs/installation/index.md:28]: With the word "light users", I meant the people who want to try out Autoware quickly without setting up the development environment.

So like this? (I know `quickly` is a bit ambiguous here. :thinking: )

```suggestion
It is recommended for beginners, people who want to try out Autoware quickly, and people who don't use Ubuntu.
```

File: docs/installation/index.md
Comment (shmpwk) [reply docs/installation/index.md:31]: I think either would be fine, but it might be easier to understand as it is now. 
autoware-documentation installation page should include step by step installation, and detailed information don't have to be included there. 
Also, people who just take a glance at docker readme in github https://github.com/autowarefoundation/autoware/tree/main/docker might need some information.

File: docs/installation/index.md
Comment (shmpwk) [reply docs/installation/index.md:28]: I found that "light users" is made in Japan English.
The correct idiom is "casual users" who don't use much autoware.
So for me, I would be add "casual users" instead.

File: docs/installation/index.md
Comment (shmpwk) [reply docs/installation/index.md:39]: StricIy speaking, "support page" is better than troubleshooting page https://autowarefoundation.github.io/autoware-documentation/main/support/,  and I've fixed it.

Comment (shmpwk) [reply docs/installation/index.md:39]: https://github.com/autowarefoundation/autoware-documentation/pull/104/commits/38068a960e1ced9ee0c0aebe2a9bb83980075d51

File: docs/installation/index.md
Comment (shmpwk) [reply docs/installation/index.md]: fixed by https://github.com/autowarefoundation/autoware-documentation/pull/104/commits/3c719259a286a93a3337cef6e1740ce7b35dcf4a

File: docs/installation/index.md
Comment (kenji-miyake) [inline docs/installation/index.md:21]: Hmm, I feel it seems a bit ambiguous with just "Autoware". :thinking:
@LalithVipulananthan @shmpwk Is it okay for you? 

```md
# Installation
## Target platforms
## Autoware
## Related tools
```

Comment (shmpwk) [reply docs/installation/index.md:21]: The changing comes from here.
https://github.com/autowarefoundation/autoware-documentation/pull/104#discussion_r873280983

Looking at ROS2 documentation, "Installing hoge" might be easier to understand. :thinking:
Screenshot from 2022-05-16 18-47-56

Comment (kenji-miyake) [reply docs/installation/index.md:21]: Looks good!

Comment (shmpwk) [reply docs/installation/index.md]: Fixed by  https://github.com/autowarefoundation/autoware-documentation/pull/104/commits/f8b939b625e3ea9d19ae5706298549cb78272082

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**
